onnxruntime/contrib_ops/cpu/attnlstm/uni_dir_attn_lstm.cc

```diff
@@ -270,15 +270,15 @@ void UniDirectionalAttnLstm<T>::Compute(const gsl::span<const T>& inputs_arg,
   // logic errors causing bounds violations.
   span_T_iter C_prev_end = batched_internal_state_prev_one_step.end();
   span_T_iter C_prev_clipped_end = batched_internal_state_clipped_one_step.end();
-  span_T_const_iter previous_state_end = batched_hidden_state_one_step.end();
+  auto previous_state_end = batched_hidden_state_one_step.end();
```

Comment:
MS GSL span doesn't have const_iterator.

note that std::span doesn't either.

Reply:
also, gsl-lite's span's iterator/const_iterator types are actually pointers. MS GSL's span's iterator is a separate type.
https://github.com/gsl-lite/gsl-lite/blob/4720a2980a30da085b4ddb4a0ea2a71af7351a48/include/gsl/gsl-lite.hpp#L3664-L3665
https://github.com/microsoft/GSL/blob/10df83d292bf5bbdc487e57dc8c2dc8c7a01f4d1/include/gsl/span#L433

Reply:
this is also why googletest was updated in this change. the original version was using `const_iterator`.